src/pages/recurring-invoices/routes.tsx

```diff
@@ -82,7 +82,10 @@ export const recurringInvoiceRoutes = (
           <Guard
             guards={[
               enabled(ModuleBitmask.RecurringInvoices),
-              permission('view_recurring_invoice'),
+              or(
+                permission('edit_recurring_invoice'),
+                assigned('/api/v1/recurring_invoices/:id')
+              ),
```

Comment:
We swapped edit for view here?

Reply:
@beganovich Actually, all actions, including those that navigate to the PDF page, are removed if the user does not have edit permission, did not create the entity, or is not an admin or owner. 

Regarding the approach, I think if the user only has view permission, they can look at the PDF preview at the bottom of the page and see the same content as on the PDF page. However, if, for example, the PDF pages contain some actions, we will need to additionally hide them and implement more logic. This way, we simply remove all actions if the user does not have at least edit permission.

So, yeah, that is the main reason for this change: removing access to the page if the user does not have access to the actual action.